scanpy_scripts/lib/_scrublet.py

```diff
@@ -19,8 +21,38 @@ def scrublet(adata, adata_sim=None, filter=False, export_table=None, **kwargs):
     if adata_sim:
         adata_sim = sc.read(adata_sim)
 
-    sce.pp.scrublet(adata, adata_sim=adata_sim, **kwargs)
-
+    # Scrublet shouldn't be run on multi-batch data, so we run the batches
+    # separately and copy the stats back to the input object
+
+    alldata = []
+    if batch_key is not None:
+        if batch_key in adata.obs.keys():
+            print("batch key %s is in obs" % batch_key)
+        else:
+            print("batch key %s is NOT in obs" % batch_key)
```

Comment:
Shouldn't we either exit here with an error or skip the batch based mode and do scrublet as normal? Otherwise next pandas statement will likely fail with a stacktrace that might be harder to debug and dilute your message above.

Reply:
I also wouldn't print the first message of the if, when the key is found, but that is a matter of taste. These days I try to reduce how much goes into the logs of everything.

Reply:
Thanks for the catch @pcm32 - this was actually rogue debugging code, but it should raise an error- done now.